Sources/SwiftSyntax/Syntax.swift

```diff
@@ -81,6 +81,76 @@ extension Syntax: CustomReflectable {
   }
 }
 
+extension Syntax: CustomDebugStringConvertible {
+  public var debugDescription: String {
+    return debugDescription()
+  }
+
+  public func debugDescription(includeChildren: Bool = true, includeTrivia: Bool = false,
+                               converter: SourceLocationConverter? = nil, mark: Syntax? = nil,
+                               indent: Int = 0) -> String {
+    var str = ""
+    debugWrite(to: &str, includeChildren: includeChildren, includeTrivia: includeTrivia, converter: converter, mark: mark, indent: indent)
+    return str
+  }
+
+  public func debugWrite<Target: TextOutputStream>(to target: inout Target,
+                                                   includeChildren: Bool = true, includeTrivia: Bool = false,
+                                                   converter: SourceLocationConverter? = nil, mark: Syntax? = nil,
+                                                   indent: Int = 0) {
+    var topLevelConverter = converter
+    if topLevelConverter == nil, let file = `as`(SourceFileSyntax.self) {
```

Comment:
Maybe this would read better as
```suggestion
    if topLevelConverter == nil, let file = self.as(SourceFileSyntax.self) {
```

Similar below. I don’t really like the look if the escaped `as`